### PR TITLE
Fixes Wormhole Generator Lag

### PR DESCRIPTION
**Pull request by Fox-McCloud**
This is hard to pin down, exactly, but there's something about the wormhole generator that can, on some instance, generate inordinate amounts of lag.

This is the only thing I can think of that would help it out.

### CODE REVIEW

File: code/game/mecha/equipment/tools/tools.dm [hunk 1]
```diff
@@ -403,11 +403,9 @@
 
 
 	action(atom/target)
-		if(!action_checks(target) || (src.loc.z in config.admin_levels)) return
-		var/list/theareas = list()
-		for(var/area/AR in orange(100, chassis))
-			if(AR in theareas) continue
-			theareas += AR
+		if(!action_checks(target) || src.loc.z == ZLEVEL_CENTCOMM)
```
Comment (PJB3005) [inline code/game/mecha/equipment/tools/tools.dm]: So, why are you making the z level check less correct?

Comment (PJB3005) [inline code/game/mecha/equipment/tools/tools.dm]: Also `src.`

**Comment by Krausus** [comment]
I took a peek at how the wormhole generator actually picks a target turf, and hoo *boy* is it bad. It picks a nearby area, and then it builds a list of every single non-obstructed turf, in that area, on the same Z-level. To do this, it iterates through *every single turf* in the picked area. Want to guess how many turfs the "space" area has?

Interestingly, wizarditis [finds a nearby turf in a very similar way](https://github.com/ParadiseSS13/Paradise/blob/78f53553f8d81ddd96d8febc52f6ee9c263cd6dc/code/datums/diseases/wizarditis.dm#L89-L113), but explicitly excludes the space area. This should either do the same (which would probably be a buff to the wormhole generator), or it should iterate through turfs in range to build an area-turf list from them directly.

**Comment by Fox-McCloud** [comment]
Yeah, Just noticed that, thanks for bringing it up @Krausus 

Going to close this for now.

**Pull request closed without merging**